lib/matplotlib/collections.py

```diff
@@ -1254,6 +1256,142 @@ def set_verts_and_codes(self, verts, codes):
+            case "x":
+                return "y"
+            case "y":
+                return "x"
+            case _:
+                raise ValueError(f"ind_dir must be 'x' or 'y', got '{self.ind_dir}'")
+
+    def set_data(
+            self, ind, dep1, dep2=0,
+            where=None, interpolate=False, step=None, **kwargs):
+        axes = getattr(self, "axes", None)
+        kwargs = self._normalise_kwargs(axes, **kwargs)
+        polys = self._make_verts(
+            ind, dep1, dep2, where, interpolate, step, axes, **kwargs)
+        self.set_verts(polys)
+
+    def _make_verts(self, ind, dep1, dep2, where, interpolate, step, axes, **kwargs):
+        dirs = (self.ind_dir, self.dep_dir, self.dep_dir)
+        arrays = (ind, dep1, dep2)
+        # Handle united data, such as dates
+        if axes:
+            arrays = tuple(axes._process_unit_info([*zip(dirs, arrays)], kwargs))
+        ind, dep1, dep2 = arrays = tuple(map(ma.masked_invalid, arrays))
+
+        names = (d + s for d, s in zip(dirs, ("", "1", "2")))
+        for name, array in zip(names, arrays):
+            if array.ndim > 1:
+                raise ValueError(f"{name!r} is not 1-dimensional")
+
+        if where is None:
+            where = True
+        else:
+            where = np.asarray(where, dtype=bool)
+            if where.size != ind.size:
+                raise ValueError(f"where size ({where.size}) does not match "
+                                 f"{self.ind_dir} size ({ind.size})")
+        where = where & ~functools.reduce(
+            np.logical_or, map(np.ma.getmaskarray, arrays))
+
+        ind, dep1, dep2 = arrays = np.broadcast_arrays(
+            np.atleast_1d(ind), dep1, dep2, subok=True)
+
+        polys = [
+            pts for idx0, idx1 in cbook.contiguous_regions(where)
+            if (pts := self._make_pts(*arrays, idx0, idx1, step, interpolate))
+            is not None]
+
+        self._pts = self._normalise_pts(np.vstack([
+            np.hstack([ind[where, None], dep[where, None]]) for dep in (dep1, dep2)]))
+
+        if "transform" in kwargs:
+            self._up_xy = kwargs["transform"].contains_branch_seperately
+
+        return polys
+
+    def _normalise_kwargs(self, axes, **kwargs):
+        if not mpl.rcParams["_internal.classic_mode"]:
+            kwargs = cbook.normalize_kwargs(kwargs, Collection)
+            if axes and not any(c in kwargs for c in ("color", "facecolor")):
+                kwargs["facecolor"] = axes._get_patches_for_fill.get_next_color()
+
+        return kwargs
+
+    def _make_pts(self, ind, dep1, dep2, idx0, idx1, step, interpolate):
+        indslice = ind[idx0:idx1]
+        dep1slice = dep1[idx0:idx1]
+        dep2slice = dep2[idx0:idx1]
+        if step is not None:
+            step_func = cbook.STEP_LOOKUP_MAP["steps-" + step]
+            indslice, dep1slice, dep2slice = \
+                step_func(indslice, dep1slice, dep2slice)
+
+        if not len(indslice):
+            return None
```

Comment:
From the source code of `cbook.contiguous_regions` and `cbook.STEP_LOOKUP_MAP`, I do not think these lines can be reached.

Reply:
130a631f77f7bc7136096ba8b04c6081e2a62db5